classes/manager.php

```diff
@@ -51,14 +54,29 @@ public static function is_flag_set(string $flag): bool {
                 isset($_GET[$flag]);
     }
 
+    /**
+     * Checks flame on/off flags and sets the session value.
+     *
+     * @return bool True if we have flame all set.
+     */
+    protected static function is_flame_all(): bool {
+        if (self::is_flag_set(self::FLAME_OFF_PARAM_NAME)) {
+            unset($_SESSION[self::FLAME_ON_PARAM_NAME]);
```

Comment:
please use the global $SESSION->foo = 'bar'; instead of $_SESSION which is the moodle code standard way of doing things